admin-dev/themes/new-theme/js/components/grid.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+const $ = window.$;
+
+/**
+ * Class is responsible for handling Grid events
+ */
+export default class Grid {
+  /**
+   * Grid's selector
+   *
+   * @param gridSelector
+   */
+  constructor(gridSelector) {
+    this.$grid = $(gridSelector);
+  }
+
+  /**
+   * Initialize grid events
+   */
+  init() {
+    this._handleBulkActionSelectAllCheckbox();
+    this._handleBulkActionCheckboxSelect();
+    this._handleCommonGridActions();
+  }
+
+  /**
+   * Handles most common grid actions (show sql, refresh list & etc.)
+   *
+   * @private
+   */
+  _handleCommonGridActions() {
+    const $gridPanel = this.$grid.closest('.js-grid-panel');
+    if (0 === $gridPanel.length) {
+      // if grid is not within a panel
+      // then grid actions are not available
+      return;
+    }
+
+    let identifier = this.$grid.attr('id');
+    let commonActionSuffix = '#' + identifier + '_action_';
+
+    let refreshListActionId = commonActionSuffix + 'ps_refresh_list';
+    let showSqlActionId = commonActionSuffix + 'ps_show_query';
+    let exportSqlManagerActionId = commonActionSuffix + 'ps_export_sql_manager';
+
+    $gridPanel.on('click', refreshListActionId, () => this._onRefreshClick());
```

Comment:
could there be a better way to handle most common grid actions?

Reply:
I ... don't know. For now, let's make it works, then we can make it easy, then we can make it pretty :)

As it's noted as "private", we can redo it when needed.